core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -337,6 +338,8 @@ object KafkaConfig {
   val SslClientAuthProp = SslConfigs.SSL_CLIENT_AUTH_CONFIG
 
   /** ********* SASL Configuration ****************/
+  val SaslMechanismProp = SaslConfigs.SASL_MECHANISM
```

Comment:
@ijuma Agree. I suppose it depends on whether SASL mechanism is treated as a property of the SASL protocol or a higher level protocol itself. There are other instances of client-side properties (eg. service name) which don't have an "inter.broker" prefix when used in the broker. I am happy with either, so will wait and see what @junrao thinks.